contracts/BatchExchange.sol

```diff
@@ -577,13 +578,10 @@ contract BatchExchange is EpochTokenLocker {
     }
 
     /** @dev determines if value is better than currently and updates if it is.
-      * @param newObjectiveValue proposed value to be updated if greater than current.
+      * @param newObjectiveValue proposed value to be updated if a great enough improvement on the current objective value
       */
     function checkAndOverrideObjectiveValue(uint256 newObjectiveValue) private {
-        require(
-            newObjectiveValue.mul(IMPROVEMENT_DENOMINATOR) > getCurrentObjectiveValue().mul(IMPROVEMENT_DENOMINATOR + 1),
-            "New objective doesn't sufficiently improve current solution"
-        );
+        require(checkProposedObjectiveValue(newObjectiveValue), "New objective doesn't sufficiently improve current solution");
```

Comment:
naming nit: here we are not checking the proposed objective value but the calculated one.

I'd also slightly prefer a method that starts with `is` (e.g. `isGoodEnoughSolution`) or the like as it conveys in the name that the function returns a bool (whereas check could also mean it just asserts it).

Reply:
Changed it.